pygmt/base_plotting.py

```diff
@@ -994,3 +994,358 @@ def text(
+            - ``"partial"`` — *strike1, dip1, strike2, fault_type, magnitude*
+            - ``"principal_axis"`` — *t_exponent, t_azimuth, t_plunge,
+              n_exponent, n_azimuth, n_plunge, p_exponent, p_azimuth, p_plunge,
+              exponent*
+
+        scale: str
+            Adjusts the scaling of the radius of the beachball, which is
+            proportional to the magnitude. Scale defines the size for
+            magnitude = 5 (i.e. scalar seismic moment M0 = 4.0E23 dynes-cm)
+        lon: int or float or list
+            Longitude(s) of event location. Ignored if `spec` is not a
+            dictionary. List must be the length of the number of events.
+            Ignored if `spec` is a DataFrame and contains a 'lon' column.
+        lat: int or float or list
+            Latitude(s) of event location. Ignored if `spec` is not a
+            dictionary. List must be the length of the number of events.
+            Ignored if `spec` is a DataFrame and contains a 'lat' column.
+        depth: int or float or list
+            Depth(s) of event location in kilometers. Ignored if `spec` is
+            not a dictionary. List must be the length of the number of events.
+            Ignored if `spec` is a DataFrame and contains a 'depth' column.
+        convention: str
+            ``"aki"`` (Aki & Richards), ``"gcmt"`` (global CMT), ``"mt"``
+            (seismic moment tensor), ``"partial"`` (partial focal mechanism),
+            or ``"principal_axis"`` (principal axis). Ignored if `spec` is a
+            dictionary or dataframe.
+        component: str
+            The component of the seismic moment tensor to plot. ``"full"`` (the
+            full seismic moment tensor), ``"dc"`` (the closest double couple
+            with zero trace and zero determinant), ``"deviatoric"`` (zero
+            trace)
+        plot_lon: int or float or list
+            Longitude(s) at which to place beachball, only used if `spec` is a
+            dictionary. List must be the length of the number of events.
+            Ignored if `spec` is a DataFrame and contains a 'plot_lon' column.
+        plot_lat: int or float or list
+            Latitude(s) at which to place beachball, only used if `spec` is a
+            dictionary. List must be the length of the number of events.
+            Ignored if `spec` is a DataFrame and contains a 'plot_lat' column.
+        offset: bool or str
+            Offsets beachballs to the longitude, latitude specified in
+            the last two columns of the input file or array, or by `plot_lon`
+            and `plot_lat` if provided. A small circle is plotted at the
+            initial location and a line connects the beachball to the circle.
+            Specify pen and optionally append ``+ssize`` to change the line
+            style and/or size of the circle.
+        {J}
+        {R}
+        {B}
+        """
+
+        # Check the spec and parse the data according to the specified
```

Comment:
I haven't read the codes, but please keep in mind that, another module [coupe](https://docs.generic-mapping-tools.org/latest/supplements/seis/coupe.html), which plots beachballs on cross-sections, will use exactly the same codes here, since **meca** and **coupe** share the same input file and **-S** option. 

We definitely don't want to maintain two copies of the same codes.

Reply:
Thanks for the tip. I haven't looked at coupe but I will.

Reply:
Good points. I'm not familiar with beachball plots, but there's a similar story going on with the `velo` function PR at #525 which shares a lot of the same implementation code as `plot` and even `plot3d` at #471. It might be the case that we'll end up having a 'core' function of some sort, and just change the module name (e.g. `meca` or `coupe`, `velo` or `plot` or `plot3d`).

For the sake of this PR though, I think it's fine to leave it like so for now. The most important thing is to have comprehensive unit tests, and we can refactor the code later (while keeping the tests the same) when `coupe` is wrapped.